src/app/space/create/deployments/apps/deployment-card.component.spec.ts

```diff
@@ -73,67 +82,158 @@ class FakeDeploymentDetailsComponent {
+
+        let menu = fixture.debugElement.query(By.css('.dropdown-menu'));
+        menuItems = menu.queryAll(By.css('li'));
+        let item = getItemByLabel('Open Application');
+        expect(item).toBeFalsy();
+      }));
+
+      it('should invoke service \'delete\' function on Delete item click', fakeAsync(() => {
+        let item = getItemByLabel('Delete');
+        expect(item).toBeTruthy();
+        expect(mockSvc.deleteApplication).not.toHaveBeenCalled();
+        item.query(By.css('a')).triggerEventHandler('click', null);
+
+        fixture.detectChanges();
+
+        expect(mockSvc.deleteApplication).toHaveBeenCalled();
+        expect(notifications.message).toHaveBeenCalled();
+      }));
+    });
+
+    describe('inactive environment', () => {
+      it('should not display', fakeAsync(() => {
+        active.next(false);
+        fixture.detectChanges();
+
+        expect(component.active).toBeFalsy();
+      }));
+    });
+  });
+
+describe('DeploymentCardComponent', () => {
+  type Context =  TestContext<DeploymentCardComponent, HostComponent>;
+  let active: Subject<boolean>;
   let mockSvc: jasmine.SpyObj<DeploymentsService>;
   let notifications: any;
-  let active: Subject<boolean>;
 
   beforeEach(fakeAsync(() => {
     active = new BehaviorSubject<boolean>(true);
-
     mockSvc = createMock(DeploymentsService);
-    mockSvc.getVersion.and.returnValue(Observable.of('1.2.3'));
-    mockSvc.getCpuStat.and.returnValue(Observable.of({ used: 1, quota: 2 }));
-    mockSvc.getMemoryStat.and.returnValue(Observable.of({ used: 3, quota: 4, units: 'GB' }));
-    mockSvc.getAppUrl.and.returnValue(Observable.of('mockAppUrl'));
-    mockSvc.getConsoleUrl.and.returnValue(Observable.of('mockConsoleUrl'));
-    mockSvc.getLogsUrl.and.returnValue(Observable.of('mockLogsUrl'));
-    mockSvc.deleteApplication.and.returnValue(Observable.of('mockDeletedMessage'));
-    mockSvc.isApplicationDeployedInEnvironment.and.returnValue(active);
-
+    initMockSvc(mockSvc, active);
     notifications = jasmine.createSpyObj<NotificationsService>('NotificationsService', ['message']);
 
-    TestBed.configureTestingModule({
-      declarations: [
-        DeploymentCardComponent,
-        FakeDeploymentsDonutComponent,
-        FakeDeploymentGraphLabelComponent,
-        FakeDeploymentDetailsComponent,
-        FakeDeploymentStatusIconComponent
-      ],
-      imports: [ BsDropdownModule.forRoot(), CollapseModule.forRoot(), ChartModule ],
-      providers: [
-        BsDropdownConfig,
-        { provide: NotificationsService, useValue: notifications },
-        { provide: DeploymentsService, useValue: mockSvc }
-      ]
-    });
-
-    fixture = TestBed.createComponent(DeploymentCardComponent);
-    component = fixture.componentInstance;
+    flush();
+    flushMicrotasks();
+  }));
 
+  initContext(DeploymentCardComponent, HostComponent, {
+    imports: [ BsDropdownModule.forRoot(), CollapseModule.forRoot(), ChartModule ],
```

Comment:
Same thing, please split this array with newlines. Also the object keys are in a different order - doesn't really matter though.